src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java

```diff
@@ -434,13 +434,35 @@ protected void writeStringToFile( String data, File file, String encoding )
         throws MojoExecutionException
     {
 
-        OutputStream outStream = null;
-        Writer writer = null;
+         OutputStream outStream = null;
         try
         {
+            byte[] binaryData = data.getBytes( encoding );
+
+            if ( file.isFile() && file.canRead() && file.length() == binaryData.length )
+            {
+                InputStream inputStream = null;
+                try
+                {
+                    byte[] buffer = new byte[ binaryData.length ];
+                    inputStream = new FileInputStream( file );
+                    inputStream.read( buffer );
+                    if ( Arrays.equals( buffer, binaryData ) ) {
```

Comment:
Hm, not sure about this. I would have (most likely) used `MessageDigest` with `SHA-1`.

@khmarbaise What's your take on this?

Reply:
The computation would be the same, and we'd have to read the whole file anyway, so using a message digest does not really change anything, apart from being more cpu intensive imho.  The can only be done because we compare pom files, which are kinda limited in size to a few kilobytes.

Reply:
Sounds reasonable.

Reply:
`Files.mismatch(file1, file2) == -1` ?
Can you give a rationale why all this is needed? What is the problem if that file is overwritten?

Reply:
@hohwille `Files.mismatch` is new in JDK 12.  I don't think the requirement is on JDK 12 atm.